### PR TITLE
Upgrade Next.js Starter Kit to use `/app` directory

### PR DESCRIPTION
**Pull request by CTNicholas**
I've upgraded our starter kit to use the latest Next.js features, including `/app` and server actions.

**Server components** have shrunk bundle sizes, including the main file shared by all pages: **157kb** → **84kb** (53%).

**Layouts** allow fetching initial dashboard data just once, and add a static header across pages.

**Server actions** have allowed me to:
- Remove 12 of the 13 API endpoints (the final one is required by NextAuth).
- Reuse the same functions on client and server.
- Remove a huge amount of types that were only necessary for synching between client and server.
- Remove a number of files that were necessary for creating reusable functions (see diagram). 

image

This is now so much simpler to use, and it's dramatically cut the size of the project.  

---

**TODO**

- [x] Upgrade to `/app` directory
  - [x] Document page
    - [x] Whiteboard
    - [x] Text editor
    - [x] Document header
    - [x] Share menu
  - [x] Dashboard page
    - [x] Basic setup
    - [x] Create document
    - [x] Pagination
    - [x] Groups/drafts
  - [x] Notifications
  - [x] Marketing page
- [x] Upgrade to NextAuth 5
- [x] Upgrade to SWR 2

**OTHER TODO**
- [x] Check guide and edit if needed
- [x] Make `create-liveblocks-app` work
- [x] Change marketing page as client/server functions no longer needed with `"use server"`
  - https://github.com/liveblocks/liveblocks.io/pull/2051

### CODE REVIEW

File: starter-kits/nextjs-starter-kit/auth.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+// More info: https://next-auth.js.org/configuration/options#secret
+export const NEXTAUTH_SECRET = process.env.NEXTAUTH_SECRET;
+
+export const {
+  handlers: { GET, POST },
+  auth,
+  signIn,
+  signOut,
+} = NextAuth({
+  secret: NEXTAUTH_SECRET || "p49RDzU36fidumaF7imGnzyhRSPWoffNjDOleU77SM4=",
```
Comment (marcbouchenoire) [inline starter-kits/nextjs-starter-kit/auth.ts:15]: Maybe a comment to explain the fallback value?

File: starter-kits/nextjs-starter-kit/primitives/Popover/Popover.module.css [hunk 2]
```diff
@@ -29,7 +29,6 @@
   border-radius: var(--radius);
   box-shadow: var(--shadow-xl);
   animation: 0.15s ease-in-out;
-  overflow: hidden;
```
Comment (marcbouchenoire) [inline starter-kits/nextjs-starter-kit/primitives/Popover/Popover.module.css:32]: Copy-paste conflict, this was added in https://github.com/liveblocks/liveblocks/pull/1437 and shouldn't be removed.

File: starter-kits/nextjs-starter-kit/liveblocks.server.config.ts [hunk 3]
```diff
@@ -1,12 +1,23 @@
-// Liveblocks API base url
-import { getProviders } from "next-auth/react";
+"server-only";
```
Comment (marcbouchenoire) [inline starter-kits/nextjs-starter-kit/liveblocks.server.config.ts:1]: Does that work? I thought it was a separate package (`import "server-only"`).

**Review by marcbouchenoire — CHANGES_REQUESTED** [review]
Nice! 🥳 I left a few comments, only small things.

**Comment by CTNicholas** [comment]
Thanks for the recommendations @marcbouchenoire! Implemented them.

**Review by marcbouchenoire — APPROVED** [review]

**Pull request merged**